### PR TITLE
[refactor][fix](evs/volume): Add multierror validation and delete expired param setting

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The `source_vol_id` was expired, It will make acc test panic, but in read function, the validation of data settings is missing.

```
panic: Invalid address to set: []string{"source_vol_id"}
2021-09-09T19:26:09.393+0800 [DEBUG] plugin.terraform-provider-huaweicloud: 
2021-09-09T19:26:09.393+0800 [DEBUG] plugin.terraform-provider-huaweicloud: goroutine 27 [running]:
2021-09-09T19:26:09.393+0800 [DEBUG] plugin.terraform-provider-huaweicloud: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc000b98500, 0x16f54db, 0xd, 0x148d480, 0x2167f60, 0x0, 0x0)
2021-09-09T19:26:09.393+0800 [DEBUG] plugin.terraform-provider-huaweicloud:     /home/lance/go/src/github.com/huaweicloud/terraform-provider-huaweicloud/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go:230 +0x371
```

**Which issue this PR fixes**:
```
NONE
```

**Special notes for your reviewer**:

**Release note**:

```release-note
1. delete the expired source volume id.
2. use multierror to verify data settings. 
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccEvsVolume_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccEvsVolume_basic -timeout 360m -parallel 4
=== RUN   TestAccEvsVolume_basic
=== PAUSE TestAccEvsVolume_basic
=== CONT  TestAccEvsVolume_basic
--- PASS: TestAccEvsVolume_basic (101.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       101.835s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_evs_volume.go [hunk 1]
```diff
@@ -214,50 +215,35 @@ func resourceEvsVolumeV3Create(d *schema.ResourceData, meta interface{}) error {
...
-	}
-	if value, ok := v.VolumeImageMetadata["image_id"]; ok {
-		d.Set("image_id", value)
-	}
-	d.Set("multiattach", v.Multiattach)
-	// set tags
+func setEvsVolumeTags(d *schema.ResourceData, resp *volumes.Volume) error {
 	tags := make(map[string]string)
-	for key, val := range v.Tags {
+	for key, val := range resp.Tags {
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_evs_volume.go]: the type of resp.Tags is map[string]string, can we set `tags` directly?

File: huaweicloud/resource_huaweicloud_evs_volume.go [hunk 2]
```diff
@@ -268,6 +253,41 @@ func resourceEvsVolumeV3Read(d *schema.ResourceData, meta interface{}) error {
...
+		d.Set("enterprise_project_id", resp.EnterpriseProjectID),
+		d.Set("region", GetRegion(d, config)),
+		d.Set("wwn", resp.WWN),
+		d.Set("multiattach", resp.Multiattach),
+		setEvsVolumeDeviceType(d, resp),
+		setEvsVolumeImageId(d, resp),
+		setEvsVolumeTags(d, resp),
+		setEvsVolumeAttachment(d, resp),
+	)
+	if err := mErr.ErrorOrNil(); err != nil {
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_evs_volume.go]: `err` has been declared above and can be used directly here.

File: huaweicloud/resource_huaweicloud_evs_volume.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/resource_huaweicloud_evs_volume.go]: absolutely yes.

File: huaweicloud/resource_huaweicloud_evs_volume.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/resource_huaweicloud_evs_volume.go]: done

**Review by chengxiangdong — COMMENTED** [review]
/lgtm

**Review by ShiChangkuo — APPROVED** [review]

**Pull request merged**